src/session/store/SessionStore.ts

```diff
@@ -21,65 +14,55 @@ export class SessionStore {
         this.redisWrapper = new RedisWrapper(redis);
     }
 
-    public load = (cookie: Cookie): EitherAsync<Failure, ISession> => {
-
-        const decode = wrapFunction<Failure, ISession, string>(Encoding.decode);
-
-        return wrapValue<Failure, Cookie>(cookie)
-            .map(_ => _.sessionId)
-            .chain(this.redisWrapper.get)
-            .chain(decode);
+    public load = async (cookie: Cookie): Promise<ISession> => {
+        return Encoding.decode(await this.redisWrapper.get(cookie.sessionId));
     };
 
-    public store = (cookie: Cookie, value: ISession, timeToLiveInSeconds: number = 3600): EitherAsync<Failure, string> => {
+    public store = async (cookie: Cookie, value: ISession, timeToLiveInSeconds: number = 3600): Promise<void> => {
         value[SessionKey.Expires] = getSecondsSinceEpoch() + timeToLiveInSeconds;
         return this.redisWrapper.set(cookie.sessionId, Encoding.encode(value), timeToLiveInSeconds);
     };
 
-    public delete = (cookie: Cookie): EitherAsync<Failure, number> => {
+    public delete = async (cookie: Cookie): Promise<void> => {
         return this.redisWrapper.del(cookie.sessionId);
     };
-
 }
 
 class RedisWrapper {
 
     public constructor(private readonly client: Redis) { }
 
-    public set = (key: string, value: string, timeToLiveInSeconds: number): EitherAsync<Failure, string> => {
-
-        const promise = this.client.set(key, value, "EX", timeToLiveInSeconds)
-            .then(r => Right(r))
-            .catch(err => Left(Failure(StoringError(err, key, value))));
+    public set = async (key: string, value: string, timeToLiveInSeconds: number): Promise<void> => {
 
-        return wrapPromise<Failure, string>(promise);
+        return this.client.set(key, value, "EX", timeToLiveInSeconds)
```

Comment:
what is "EX" supposed to mean? maybe create a const to store this value for readability?

Reply:
EX defines expiry mode for Redis client. It is so buried (wrapper class is not exposed) and so insignificant that I do not think adding constant provides much value here.